premium/monitor_test.go

```diff
@@ -10,7 +10,7 @@ import (
 )
 
 type quotaResponse struct {
-	hasQuota bool
+	hasQuota QuotaCheckResult
```

Comment:
```suggestion
	quotaCheckResult QuotaCheckResult
```

Might as well rename the field too